### PR TITLE
fix: escape invalid xml characters

### LINKED ISSUE #519: Escape invalid xml characters for all listing APIs
Currently Next Marker, Marker, Prefix and CommonPrefixes keys are not escaped in the xml. This results in invalid XML if object keys have invalid XML characters.

### PR DESCRIPTION
**Pull request by rahulreddy**
fixes #519

### CODE REVIEW

**Comment by rahulreddy** [comment]
@ironman-machine

**Comment by rahulreddy** [comment]
@ironman-machine try

File: tests/functional/aws-node-sdk/test/bucket/get.js [hunk 1]
```diff
@@ -349,5 +348,104 @@ describe('GET Bucket - AWS.S3.listObjects', () => {
...
+                    .catch(done);
+            });
+        });
+
+        it('should list objects with special chars in CommonPrefixes', done => {
+            const s3 = bucketUtil.s3;
+            const Bucket = bucketName;
+            const objects = [{ Bucket, Key: '&amp#' },
+            { Bucket, Key: ';quot#' }, { Bucket, Key: '\'apos#' },
+            { Bucket, Key: '<lt#' }, { Bucket, Key: '<lt#' },
```
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/bucket/get.js]: Some duplicate objects here

Comment (rahulreddy) [inline tests/functional/aws-node-sdk/test/bucket/get.js]: Thanks. Fixed them.

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by vrancurel** [comment]
can we merge ?

**Comment by rahulreddy** [comment]
Yes

**Comment by LaurenSpiegel** [comment]
isn't there going to be the same problem with mpu listing and other api's?

**Comment by rahulreddy** [comment]
I am doing a sweep across all those APIs. Currently the customer is running into issues for only this api, hence the quick fix.

**Pull request merged**